### PR TITLE
Support for N-Dimmemsional Curves

### PR DESCRIPTION
**Pull request by AlexKaracaoglu**
This is in reference to issue #14399 

The changes allow the user to:
- Create higher dimensional curves
- Get the properties of these higher dimensional curves (ambient_dimension, parameters, ...)
- Evaluate points on these higher order curves
- Scale and translate these higher order curves

In the code, there are examples of how the changed functions are to be used, and have been tested on non 2-dimensional curves

### CODE REVIEW

File: sympy/geometry/curve.py [hunk 1]
```diff
@@ -70,20 +71,53 @@ class Curve(GeometrySet):
...
+    >>> C = Curve((sin(t), cos(t), t), (t, 0, 2))
+    >>> C
+    Curve((sin(t), cos(t), t), (t, 0, 2))
+    >>> C.functions
+    (sin(t), cos(t), t)
+    >>> C.limits
+    (t, 0, 2)
+    >>> C.parameter
+    t
+    >>> C.subs(t,4)
```
Comment (normalhuman) [inline sympy/geometry/curve.py]: space after comma

File: sympy/geometry/curve.py [hunk 2]
```diff
@@ -70,20 +71,53 @@ class Curve(GeometrySet):
     Point2D(4, 16)
     >>> C.arbitrary_point(a)
     Point2D(a, a**2)
+
+    Higher Dimmension Curves:
+    >>> from sympy import sin, cos, Symbol
```
Comment (normalhuman) [inline sympy/geometry/curve.py]: No need to repeat the imports that are already made in this docstring example.

File: sympy/geometry/curve.py [hunk 3]
```diff
@@ -329,35 +424,56 @@ def rotate(self, angle=0, pt=None):
...
         >>> from sympy import pi
         >>> from sympy.abc import x
-        >>> Curve((x, x), (x, 0, 1)).scale(2)
+        >>> Curve((x, x), (x, 0, 1)).scale([2,1])
         Curve((2*x, x), (x, 0, 1))
+
+        Higher Dimmension Curves:
+        >>> from sympy.abc import x, t, s
+        >>> C = Curve((t, t, t), (t, 0, 2))
+        >>> C.scale([2,0,4])
```
Comment (normalhuman) [inline sympy/geometry/curve.py]: spaces after commas

File: sympy/geometry/curve.py [hunk 4]
```diff
@@ -1,4 +1,4 @@
-"""Curves in 2-dimensional Euclidean space.
+"""Curves in n-dimensional Euclidean space.
```
Comment (normalhuman) [inline sympy/geometry/curve.py]: Where n is 2 or 3, right?

**Review by normalhuman — COMMENTED** [review]
Tests for new functionality should be added to geometry/tests/test_curve.py

"n-dimensional" means n can be any positive integer, including, say, 5 or 42. But it seems that only 3-dimensional curves are being added here. I suggest changing "Higher Dimmension Curve" to "3-dimensional curve" throughout

**Comment by normalhuman** [comment]
See also [Development workflow](https://github.com/sympy/sympy/wiki/Development-workflow) regarding creating a separate branch for your patch, and testing the patch locally.

**Comment by woodscn** [comment]
I could probably fix the formatting issues referenced here, if that's all that's keeping this from being merged. I wouldn't be able to provide long-term support, though. @AlexKaracaoglu @normalhuman

**Pull request closed without merging**